### PR TITLE
Issue#13: Go-based operator for kAppNav.

### PR DESCRIPTION
**Pull request by mrglavas**
This PR resolves (https://github.com/kappnav/Issues/issues/13), replacing the Helm operator with a native Go-based operator. This was developed using Operator SDK v0.10.0. Expanded the README to provide details of the new operator.

### CODE REVIEW

**Comment by mrglavas** [comment]
Also worth noting that the new operator also replaces the "init" container, embedding relevant logic from that container within it. The "init" container is not installed by the operator anymore.

File: .code-workspace.code-workspace [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (cvignola) [inline .code-workspace.code-workspace]: Sure you want this under git control?  Maybe belongs in .gitignore

File: build.sh [hunk 2]
```diff
@@ -6,11 +6,14 @@ if [ -e ../build/version.sh ]; then
...
 
+CURRENT=`pwd`
+PROJECT=`basename "$CURRENT"`
+
 # Update version numbers in yaml files
 ./updateVersionsinYamls.sh
 
 echo "Building ${IMAGE} ${VERSION}"
-docker build --pull --build-arg VERSION=${VERSION} --build-arg BUILD_DATE=$(date -u +'%Y-%m-%dT%H:%M:%SZ')  -f build/Dockerfile -t ${IMAGE} .
+operator-sdk build --image-build-args "--pull --build-arg VERSION=${VERSION} --build-arg BUILD_DATE=$(date -u +'%Y-%m-%dT%H:%M:%SZ') --build-arg PROJECT_NAME=${PROJECT}" ${IMAGE}
```
Comment (cvignola) [inline build.sh]: Is it possible to run the operator-sdk inside the docker build?  What would be the pro/con of doing that?

Comment (cvignola) [reply build.sh]: How big and heavy is the SDK?

File: README.md [hunk 3]
```diff
@@ -1,14 +1,52 @@
...
 
-# kAppNav Helm Operator
+# kAppNav Operator
 
-Used to install Application Navigator using helm chart. The operator is to read from a custom object that represents the application navigator instance. No "helm install" is needed.
+Used to install Application Navigator. The operator is to read from a custom object that represents the application navigator instance. No "helm install" is needed.
 
 ## Build
 
+This project was developed with Operator SDK v0.10.0.
```
Comment (cvignola) [inline README.md]: Curious:  can you build and run this controller locally on your own machine, outside a container,  and not on Kubernetes?  I remember another controller I saw (a sample somewhere) could run locally.  Is that how you developed this?

File: .code-workspace.code-workspace (hunk 1 above)
Comment (mrglavas) [reply .code-workspace.code-workspace]: Good catch. I'll add it to .gitignore.

File: build.sh (hunk 2 above)
Comment (mrglavas) [reply build.sh]: The v0.10.0 binary looks to be about 64 MBs: https://github.com/operator-framework/operator-sdk/releases/download/v0.10.0/operator-sdk-v0.10.0-x86_64-linux-gnu.

Comment (mrglavas) [reply build.sh]: The operator-sdk build directly produces a Docker image (or Podman or Buildah image). Doesn't look like they created anything that makes it easy to build inside of a container.

File: README.md (hunk 3 above)
Comment (mrglavas) [reply README.md]: The operator's main controller loop (called reconcile) is invoked based on state changes to the CR and secondary resources it created. Need to have at least a fake Kubernetes in order to be able invoke that outside. I believe Appsody has unit tests which do just that with the Docker container. I would assume it's possible to grab the binary that the operator-sdk produced out of the temp build directory and use it directly, though I have never tried that.

Comment (mrglavas) [reply README.md]: I've got Docker desktop running on my machine with Kubernetes. Build and deploy is rather quick so I wasn't tempted to cut Docker/Kubernetes out of the equation.

File: build.sh (hunk 2 above)
Comment (mrglavas) [reply build.sh]: There's an install script (.travis/install-operator-sdk.sh) that's used by the Travis build. I didn't advertise it in the README, though I suppose developers could use it too. Apparently only took 1.6 seconds to run in the last Travis build.

**Pull request merged**